### DISCUSSION TITLE
Is there a Meson generator?

### DISCUSSION THREAD

**Discussion by ghost** [premake/premake-core#2304] [General]
This page has a CMake generator, but there is no Meson generator:

https://premake.github.io/community/modules/#build-system-modules

**Comment by KyrietS** [comment]
These are community maintained add-on modules. They exist because there are some good souls out there who made them. Unfortunately there is no generator for meson project files and if there is, I don't know about it.

What is your use-case? Why do you want Premake to generate meson configuration?

**Reply by ghost** [reply]
Nope. I have no use-case for the Meson generator. I'm only curious. Btw, Meson is popular along GNOME developers. It can generate compact ninja build script that is said to be faster than the one generated by CMake. If an existing Premake project wants to present itself to GNOME developers, I think it should better be able to be built with Meson.

**Reply by KyrietS** [reply]
Interoperability with different build systems is not something that Premake was designed for. Yes, you have CMake generator but please note that compared to Mason's CMake module, it "only" generates CMake files that you can use to generate or compile with CMake. You can't import  CMake project and mix it with a Premake's one. It's kind of different philosophy 🙃

**Comment by samsinsane** [comment]
I'm going to close this discussion as this isn't something Premake should do natively. As has already been mentioned, the module system is there if someone wants to make a premake-meson module, similar to the premake-cmake module.
